components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/loader/UIBasedConfigurationLoader.java

```diff
@@ -306,7 +308,8 @@ private void loadStepAuthenticator(StepConfig stepConfig, IdentityProvider idp,
 
         if (idp != null) {
             authenticatorConfig.getIdpNames().add(idp.getIdentityProviderName());
-            authenticatorConfig.getIdps().put(idp.getIdentityProviderName(), idp);
+            //authenticatorConfig.getIdps().put(idp.getIdentityProviderName(), idp);
+            authenticatorConfig.addIdP(idp.getIdentityProviderName(), tenantDomain);
```

Comment:
Just for my understanding, at this instance the idp object doesn't have the resource id?